src/rotator_library/http_client_pool.py

```
@@ -0,0 +1,514 @@
+            client: The client to check
+
+        Returns:
+            True if the client is closed or None, False otherwise
+        """
+        if client is None:
+            return True
+        # httpx.AsyncClient sets _client to None when closed
+        # We check the internal _client attribute which is the actual transport
+        return getattr(client, "_client", None) is None
+
+    async def _ensure_client(self, streaming: bool) -> httpx.AsyncClient:
```

Ruff (0.15.4)

[error] Boolean-typed positional argument in function definition

(FBT001)